lib/view_model/utils/collections.rb

```diff
@@ -1,21 +1,27 @@
+# frozen_string_literal: true
+
 class ViewModel::Utils
   module Collections
     def self.count_by(enumerable)
-      enumerable.each_with_object(Hash.new(0)) do |el, counts|
-        key         = yield(el)
-        counts[key] += 1 unless key.nil?
+      enumerable.each_with_object({}) do |el, counts|
+        key = yield(el)
+
+        unless key.nil?
+          counts[key] ||= 0
+          counts[key] += 1
+        end
       end
     end
 
     refine Array do
       def contains_exactly?(other)
-        vals = to_set
-        other.each { |o| return false unless vals.delete?(o) }
-        vals.blank?
+        mine   = count_by { |x| x }
```

Comment:
What's the motivation to rewrite this?

Reply:
It was incorrect when there were multiple instances of the same value.